forum/frontend/src/components/register.js

```diff
@@ -0,0 +1,120 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import { useNavigate } from 'react-router-dom';
+import '../styles/register.css';
+
+const Register = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [username, setUsername] = useState('');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [phone, setPhone] = useState('');
+  const [error, setError] = useState('');
+  const [validationError, setValidationError] = useState('');
+  const navigate = useNavigate();
+
+  const validateInputs = () => {
+    if (!username) return 'Username is required';
+    if (!email) return 'Email is required';
+    if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email)) return 'Invalid email format';
+    if (!firstName) return 'First name is required';
+    if (!lastName) return 'Last name is required';
+    if (!phone) return 'Phone number is required';
+    if (!/^\+380\d{9}$/.test(phone)) return 'Phone number must be in the format +380XXXXXXXXX (12 digits)';
```

Comment:
The validation expects only Ukrainian numbers (+380XXXXXXXXX). Use a more flexible phone validation regex or international input libraries (like `react-phone-input-2`).